pyro/nn/module.py

```diff
@@ -68,8 +93,10 @@ class PyroModule(torch.nn.Module):
         x = my_module.x  # Triggers a pyro.sample statement.
         y = my_module.y  # Triggers one pyro.sample + two pyro.param statements.
 
-    Note that, because sample statements can appear only once in a trace, you
-    can read the value of a random attribute only once per inference step.
+    Note that param and sample attribute access is cached within each
```

Comment:
+1 on this change! This was also my only concern about `PyroSample`, I think this behavior is much more intuitive.